main/lsp/LSPTypechecker.h

```diff
@@ -92,10 +92,21 @@ class LSPTypechecker final {
         Cancelable,
     };
 
+    struct SlowPathResult {
+        // True when the changes from the slow path have been committed. A false value here indicates that the slow path
+        // operation was cancelled.
+        bool committed = false;
+
+        // When the slow path was run with `SlowPathMode::Init`, this is a copy of the GlobalState that is taken right
+        // after indexing has completed. This GlobalState is suitable to be given to the indexer thread, as its name
+        // table will be populated with the current state of the codebase.
+        std::unique_ptr<core::GlobalState> indexedState;
+    };
```

Comment:
Decided to not go with `std::variant`, eh?

If we're not doing `std::variant`, can we have comments about how these two relate to each other and how `SlowPathMode` affects this?  My understanding is that we can have:

1. true + non-null `indexedState`
2. false + null `indexedState`

But it's not obvious to me if those can change based on `SlowPathMode` (I think the former can?).

Reply:
I switched over to `std::variant`, what do you think?